### PR TITLE
Update sbt-apache-sonatype and handle rename

### PR DESCRIPTION
**Pull request by mdedetrich**
Updates sbt-apache-sonatype. Note that this new version renames `SonatypeApachePlugin` to `ApacheSonatypePlugin` to be consistent with the project name.

### CODE REVIEW

**Review by pjfanning — APPROVED** [review]
lgtm

**Comment by mdedetrich** [comment]
Ill merge this PR after the Scala 3 PR is merged to prevent further conflicts

**Comment by mdedetrich** [comment]
PR has been pushed with newer version of sbt-apache-sonatype in order to include https://github.com/mdedetrich/sbt-apache-sonatype/pull/15

**Pull request merged**
